app/src/main/java/com/sun/moviedb_45/data/source/local/MovieLocalDataSource.java

```diff
@@ -1,6 +1,16 @@
 package com.sun.moviedb_45.data.source.local;
 
+import android.content.Context;
+
 import com.sun.moviedb_45.data.source.MovieDataSource;
 
 public class MovieLocalDataSource implements MovieDataSource.Local {
+    private static MovieLocalDataSource sInstance;
+
+    public static MovieLocalDataSource getInstance(Context context){
```

Comment:
nên để synchronized để đảm bảo thread-safe
```suggestion
    public static synchronized MovieLocalDataSource getInstance(Context context){
```